diff -urNp /dev/null Glide3/cvg/glide3/src/fxgasm.h
--- /dev/null
+++ Glide3/cvg/glide3/src/fxgasm.h
@@ -0,0 +1,83 @@
+
+;----------------------------------------------------------------------
+; Assembler offsets for SSTREGS struct
+;----------------------------------------------------------------------
+
+
+
+;----------------------------------------------------------------------
+; Assembler offsets for GC struct
+;----------------------------------------------------------------------
+
+base_ptr		equ 00000000h
+reg_ptr			equ 00000004h
+lfb_ptr			equ 0000000ch
+tex_ptr			equ 00000008h
+cull_mode		equ 000000d0h
+tsuDataList		equ 00000014h
+triPacketHdr	equ 00000358h
+cullStripHdr	equ 0000035ch
+paramMask	equ 00000360h
+triSetupProc	equ 00000344h
+drawTrianglesProc	equ 00000348h
+drawVertexList	equ 0000034ch
+fifoStart	equ 00000364h
+fifoEnd	equ 00000368h
+fifoOffset	equ 0000036ch
+fifoSize	equ 00000370h
+fifoJmpHdr	equ 00000374h
+fifoPtr	equ 00000378h
+fifoRead	equ 0000037ch
+fifoRoom	equ 00000380h
+roomToReadPtr	equ 00000384h
+roomToEnd	equ 00000388h
+fifoLfbP	equ 0000038ch
+lfbLockCount	equ 00000390h
+vertexSize	equ 00000278h
+vertexStride	equ 00000274h
+invalid	equ 00000280h
+CoordinateSpace	equ 0000033ch
+paramIndex	equ 000000d4h
+vp_hwidth	equ 00000208h
+vp_hheight	equ 0000020ch
+vp_hdepth	equ 00000210h
+vp_ox	equ 000001fch
+vp_oy	equ 00000200h
+vp_oz	equ 00000204h
+colorType	equ 0000027ch
+wInfo_offset	equ 00000228h
+qInfo_mode	equ 0000025ch
+qInfo_offset	equ 00000260h
+q0Info_offset	equ 00000268h
+q1Info_offset	equ 00000270h
+q0Info_mode	equ 00000264h
+q1Info_mode	equ 0000026ch
+tmu0_s_scale	equ 00000144h
+tmu0_t_scale	equ 00000148h
+tmu1_s_scale	equ 00000184h
+tmu1_t_scale	equ 00000188h
+SIZEOF_GrState		equ 00000274h
+SIZEOF_GC		equ 00000438h
+
+
+;----------------------------------------------------------------------
+; Assembler offsets for GlideRoot struct
+;----------------------------------------------------------------------
+
+p6Fencer		equ 00000000h
+current_sst		equ 00000004h
+CPUType			equ 00000008h
+curGC			equ 0000002ch
+curTriSize		equ 00000038h
+stats_linesDrawn		equ 00000098h
+stats_otherTrisDrawn		equ 000000a4h
+trisProcessed		equ 0000009ch
+trisDrawn		equ 000000a0h
+pool_f0	equ 00000044h
+pool_fHalf	equ 00000048h
+pool_f1	equ 0000004ch
+pool_f255	equ 00000050h
+pool_fTemp1	equ 00000054h
+pool_fTemp2	equ 00000058h
+SIZEOF_GlideRoot	equ 0000126ch
+
diff -urNp /dev/null Glide3/cvg/glide3/src/fxinline.h
--- /dev/null
+++ Glide3/cvg/glide3/src/fxinline.h
@@ -0,0 +1,10 @@
+#ifndef __FX_INLINE_H__
+#define __FX_INLINE_H__
+
+#define kCurGCOffset   0x2CUL
+#define kTriProcOffset 0x344UL
+#define kGCStateInvalid 0x280UL
+/* The # of 2-byte entries in the hw fog table */
+#define kInternalFogTableEntryCount 0x40
+
+#endif /* __FX_INLINE_H__ */
diff -urNp /dev/null Glide3/h3/glide3/src/fxgasm.h
--- /dev/null
+++ Glide3/h3/glide3/src/fxgasm.h
@@ -0,0 +1,80 @@
+
+;----------------------------------------------------------------------
+; Assembler offsets for SSTREGS struct
+;----------------------------------------------------------------------
+
+
+
+;----------------------------------------------------------------------
+; Assembler offsets for GC struct
+;----------------------------------------------------------------------
+
+base_ptr		equ 00000074h
+reg_ptr			equ 00000078h
+lfb_ptr			equ 00000080h
+cull_mode		equ 000001d0h
+tsuDataList		equ 00000114h
+triPacketHdr	equ 00000a60h
+cullStripHdr	equ 00000a64h
+paramMask	equ 00000a68h
+fifoStart	equ 00000a8ch
+fifoEnd	equ 00000a90h
+fifoOffset	equ 00000a94h
+fifoSize	equ 00000a98h
+fifoJmpHdr	equ 00000a9ch
+fifoPtr	equ 00000a6ch
+fifoRead	equ 00000a70h
+fifoRoom	equ 00000a74h
+roomToReadPtr	equ 00000aa4h
+roomToEnd	equ 00000aa8h
+lfbLockCount	equ 00000aach
+triSetupProc	equ 00000a4ch
+drawTrianglesProc	equ 00000a50h
+drawVertexList	equ 00000a4ch
+vertexSize	equ 00000958h
+vertexStride	equ 00000954h
+invalid	equ 00000960h
+CoordinateSpace	equ 00000a20h
+paramIndex	equ 000001d4h
+vp_hwidth	equ 000008e8h
+vp_hheight	equ 000008ech
+vp_hdepth	equ 000008f0h
+vp_ox	equ 000008dch
+vp_oy	equ 000008e0h
+vp_oz	equ 000008e4h
+colorType	equ 0000095ch
+wInfo_offset	equ 00000908h
+qInfo_mode	equ 0000093ch
+qInfo_offset	equ 00000940h
+q0Info_offset	equ 00000948h
+q1Info_offset	equ 00000950h
+q0Info_mode	equ 00000944h
+q1Info_mode	equ 0000094ch
+fogInfo_offset	equ 00000918h
+fogInfo_mode	equ 00000914h
+tmu0_s_scale	equ 00000854h
+tmu0_t_scale	equ 00000858h
+tmu1_s_scale	equ 00000870h
+tmu1_t_scale	equ 00000874h
+fbi_fbzMode	equ 000001ech
+curTriSize		equ 0000004ch
+trisProcessed		equ 0000000ch
+trisDrawn		equ 00000010h
+lostContext		equ 00009250h
+windowed		equ 00009258h
+bInfo		equ 0000010ch
+
+;----------------------------------------------------------------------
+; Assembler offsets for GlideRoot struct
+;----------------------------------------------------------------------
+
+p6Fencer		equ 00000000h
+current_sst		equ 0000000ch
+CPUType			equ 00000010h
+tlsOffset			equ 00000008h
+pool_f255	equ 0000002ch
+SIZEOF_GrState		equ 00000854h
+SIZEOF_GC		equ 00009294h
+
+SIZEOF_GlideRoot	equ 000b74fch
+
diff -urNp /dev/null Glide3/h3/glide3/src/fxinline.h
--- /dev/null
+++ Glide3/h3/glide3/src/fxinline.h
@@ -0,0 +1,10 @@
+#ifndef __FX_INLINE_H__
+#define __FX_INLINE_H__
+
+#define kTriProcOffset 0x917CUL
+/* The # of 2-byte entries in the hw fog table */
+#define kInternalFogTableEntryCount 0x40
+#define kTLSOffset 0x8UL
+#define kLostContextOffset 0x9250UL
+
+#endif /* __FX_INLINE_H__ */
diff -urNp /dev/null Glide3/h5/glide3/src/fxbldno.h
--- /dev/null
+++ Glide3/h5/glide3/src/fxbldno.h
@@ -0,0 +1,2 @@
+#define BUILD_NUMBER		40404
+#define BUILD_NUMBER_STR	"40404"
diff -urNp /dev/null Glide3/h5/glide3/src/fxgasm.h
--- /dev/null
+++ Glide3/h5/glide3/src/fxgasm.h
@@ -0,0 +1,82 @@
+
+;----------------------------------------------------------------------
+; Assembler offsets for SSTREGS struct
+;----------------------------------------------------------------------
+
+
+
+;----------------------------------------------------------------------
+; Assembler offsets for GC struct
+;----------------------------------------------------------------------
+
+base_ptr		equ 00000074h
+reg_ptr			equ 00000078h
+lfb_ptr			equ 00000080h
+cull_mode		equ 000001e0h
+tsuDataList		equ 00000124h
+triPacketHdr	equ 00000e00h
+cullStripHdr	equ 00000e04h
+paramMask	equ 00000e08h
+fifoStart	equ 00000e2ch
+fifoEnd	equ 00000e30h
+fifoOffset	equ 00000e34h
+fifoSize	equ 00000e38h
+fifoJmpHdr	equ 00000e3ch
+fifoPtr	equ 00000e0ch
+fifoRead	equ 00000e10h
+fifoRoom	equ 00000e14h
+roomToReadPtr	equ 00000e44h
+roomToEnd	equ 00000e48h
+lfbLockCount	equ 00000e4ch
+triSetupProc	equ 00000dech
+drawTrianglesProc	equ 00000df0h
+drawVertexList	equ 00000dech
+vertexSize	equ 00000b88h
+vertexStride	equ 00000b84h
+invalid	equ 00000b90h
+CoordinateSpace	equ 00000dbch
+paramIndex	equ 000001e4h
+vp_hwidth	equ 00000b18h
+vp_hheight	equ 00000b1ch
+vp_hdepth	equ 00000b20h
+vp_ox	equ 00000b0ch
+vp_oy	equ 00000b10h
+vp_oz	equ 00000b14h
+colorType	equ 00000b8ch
+wInfo_offset	equ 00000b38h
+qInfo_mode	equ 00000b6ch
+qInfo_offset	equ 00000b70h
+q0Info_offset	equ 00000b78h
+q1Info_offset	equ 00000b80h
+q0Info_mode	equ 00000b74h
+q1Info_mode	equ 00000b7ch
+fogInfo_offset	equ 00000b48h
+fogInfo_mode	equ 00000b44h
+depth_range	equ 00000a5ch
+tmu0_s_scale	equ 00000a0ch
+tmu0_t_scale	equ 00000a10h
+tmu1_s_scale	equ 00000a34h
+tmu1_t_scale	equ 00000a38h
+fbi_fbzMode	equ 00000210h
+curTriSize		equ 0000004ch
+trisProcessed		equ 0000000ch
+trisDrawn		equ 00000010h
+lostContext		equ 000096e4h
+windowed		equ 000096ech
+bInfo		equ 0000011ch
+
+;----------------------------------------------------------------------
+; Assembler offsets for GlideRoot struct
+;----------------------------------------------------------------------
+
+p6Fencer		equ 00000000h
+current_sst		equ 0000000ch
+CPUType			equ 00000020h
+tlsOffset			equ 00000008h
+pool_f255	equ 00000054h
+pool_f1	equ 00000050h
+SIZEOF_GrState		equ 00000be4h
+SIZEOF_GC		equ 00009738h
+
+SIZEOF_GlideRoot	equ 000bd618h
+
diff -urNp /dev/null Glide3/h5/glide3/src/fxinline.h
--- /dev/null
+++ Glide3/h5/glide3/src/fxinline.h
@@ -0,0 +1,10 @@
+#ifndef __FX_INLINE_H__
+#define __FX_INLINE_H__
+
+#define kTriProcOffset 0x9558UL
+/* The # of 2-byte entries in the hw fog table */
+#define kInternalFogTableEntryCount 0x40
+#define kTLSOffset 0x8UL
+#define kLostContextOffset 0x96e4UL
+
+#endif /* __FX_INLINE_H__ */
diff -urNp /dev/null Glide3/h5/incsrc/gendate.h
--- /dev/null
+++ Glide3/h5/incsrc/gendate.h
@@ -0,0 +1,4 @@
+#ifndef GENDATE_H
+#define GENDATE_H
+#define BUILD_DATE "Thu Aug 20 10:50:10 2015 (from incsrc/gendate.h)"
+#endif
diff -urNp /dev/null Glide3/sst1/glide3/src/fxgasm.h
--- /dev/null
+++ Glide3/sst1/glide3/src/fxgasm.h
@@ -0,0 +1,58 @@
+SST_CHIP_MASK equ 03c00h
+
+;----------------------------------------------------------------------
+; Assembler offsets for SSTREGS struct
+;----------------------------------------------------------------------
+
+FVAX			equ 00000088h
+FVAY			equ 0000008ch
+FVBX			equ 00000090h
+FVBY			equ 00000094h
+FVCX			equ 00000098h
+FVCY			equ 0000009ch
+FTRIANGLECMD		equ 00000100h
+
+FDPDX_OFFSET		equ 00000020h
+FDPDY_OFFSET		equ 00000040h
+
+;----------------------------------------------------------------------
+; Assembler offsets for GC struct
+;----------------------------------------------------------------------
+
+base_ptr		equ 00000000h
+reg_ptr			equ 00000004h
+lfb_ptr			equ 0000000ch
+cull_mode		equ 000003c4h
+dataList		equ 00000018h
+invalid		equ 0000055ch
+coord_space		equ 0000060ch
+color_type		equ 00000558h
+fifoFree		equ 000003c8h
+SIZEOF_GrState		equ 0000024ch
+SIZEOF_GrHwConfiguration	equ 00000094h
+SIZEOF_GC		equ 00000754h
+
+
+;----------------------------------------------------------------------
+; Assembler offsets for GlideRoot struct
+;----------------------------------------------------------------------
+
+p6Fencer		equ 00000000h
+current_sst		equ 00000004h
+CPUType			equ 00000008h
+curGC			equ 0000000ch
+curTriSize		equ 00000010h
+trisProcessed		equ 00000080h
+trisDrawn		equ 00000084h
+SIZEOF_GlideRoot	equ 00001e7ch
+
+
+;----------------------------------------------------------------------
+; Assembler offsets for dataList struct
+;----------------------------------------------------------------------
+
+dl_i			equ 00000000h
+dl_addr			equ 00000004h
+dl_bddr			equ 00000008h
+SIZEOF_dataList		equ 00000014h
+
diff -urNp /dev/null Glide3/sst1/glide3/src/fxinline.h
--- /dev/null
+++ Glide3/sst1/glide3/src/fxinline.h
@@ -0,0 +1,7 @@
+#ifndef __FX_INLINE_H__
+#define __FX_INLINE_H__
+
+/* The # of 2-byte entries in the hw fog table */
+#define kInternalFogTableEntryCount 0x40
+
+#endif /* __FX_INLINE_H__ */
