*** a\drivers\glide\wgl.c	2016-11-25 14:06:19.529875000 -0500
--- b\drivers\glide\wgl.c	2019-05-26 13:02:35.000000000 -0400
***************
*** 36,41 ****
--- 36,42 ----
      1,                     /* version number */
      PFD_DRAW_TO_WINDOW |   /* support window */
      PFD_SUPPORT_OPENGL |   /* support OpenGL */
+ 	PFD_SWAP_COPY |
      PFD_DOUBLEBUFFER,      /* double buffered */
      PFD_TYPE_RGBA,         /* RGBA type */
      16,                    /* color depth */
***************
*** 51,61 ****
      0,                     /* reserved */
      0, 0, 0                /* layer masks ignored */
  },
! { 
      sizeof(PIXELFORMATDESCRIPTOR),  /*  size of this pfd */
      1,                     /* version number */
      PFD_DRAW_TO_WINDOW |   /* support window */
      PFD_SUPPORT_OPENGL |   /* support OpenGL */
      PFD_DOUBLEBUFFER,      /* double buffered */
      PFD_TYPE_RGBA,         /* RGBA type */
      32,                    /* color depth */
--- 52,84 ----
      0,                     /* reserved */
      0, 0, 0                /* layer masks ignored */
  },
! {
!     sizeof(PIXELFORMATDESCRIPTOR),  /*  size of this pfd */
!     1,                     /* version number */
!     PFD_DRAW_TO_WINDOW |   /* support window */
!     PFD_SUPPORT_OPENGL |   /* support OpenGL */
! 	PFD_SWAP_COPY |
!     PFD_DOUBLEBUFFER,      /* double buffered */
!     PFD_TYPE_RGBA,         /* RGBA type */
!     24,                    /* color depth */
!     8, 0, 8, 8, 8, 16,     /* color bits */
!     8,                     /* alpha buffer */
!     24,                    /* shift bit */
!     0,                     /* no accumulation buffer */
!     0, 0, 0, 0,            /* accum bits ignored */
!     24,                    /* z-buffer */
!     8,                     /* stencil buffer */
!     0,                     /* no auxiliary buffer */
!     PFD_MAIN_PLANE,        /* main layer */
!     0,                     /* reserved */
!     0, 0, 0                /* layer masks ignored */
! },
! {
      sizeof(PIXELFORMATDESCRIPTOR),  /*  size of this pfd */
      1,                     /* version number */
      PFD_DRAW_TO_WINDOW |   /* support window */
      PFD_SUPPORT_OPENGL |   /* support OpenGL */
+ 	PFD_SWAP_COPY |
      PFD_DOUBLEBUFFER,      /* double buffered */
      PFD_TYPE_RGBA,         /* RGBA type */
      32,                    /* color depth */
***************
*** 166,175 ****
  {
      int i, numPixelFormats, bestIndex = -1;
      const PIXELFORMATDESCRIPTOR *ppfdBest = NULL;
  
      WINLOG((eff, "%s:\n", __FUNCTION__));
  
!     if (ppfd->dwFlags != (ppfd->dwFlags & 
  				    (
  				    PFD_DRAW_TO_WINDOW |
  				    PFD_DRAW_TO_BITMAP |
--- 189,203 ----
  {
      int i, numPixelFormats, bestIndex = -1;
      const PIXELFORMATDESCRIPTOR *ppfdBest = NULL;
+     DEVMODE dm;
+ 
+     /* FS: Choose the best PFD based on our display BPP. */
+     memset(&dm, 0, sizeof(DEVMODE));
+     EnumDisplaySettings(NULL, ENUM_CURRENT_SETTINGS, &dm);
  
      WINLOG((eff, "%s:\n", __FUNCTION__));
  
!     if (ppfd->dwFlags != (ppfd->dwFlags &
  				    (
  				    PFD_DRAW_TO_WINDOW |
  				    PFD_DRAW_TO_BITMAP |
***************
*** 263,269 ****
  	    continue;
  	}
  
! 	if (ppfd->cStencilBits && !ppfdCandidate->cStencilBits) {
  	    if (!YES("wgl.ignore.stencilbuffer")) continue;
  	}
  
--- 291,297 ----
  	    continue;
  	}
  
! 	if (ppfd->cStencilBits && !ppfdCandidate->cStencilBits && (dm.dmBitsPerPel == 32)) { /* FS: Stencil buffer only available on 32bpp */
  	    if (!YES("wgl.ignore.stencilbuffer")) continue;
  	}
  
***************
*** 271,276 ****
--- 299,309 ----
  	    continue;
  	}
  
+ 	/* FS: Choose the best PFD based on our display BPP. */
+ 	if ((dm.dmBitsPerPel > 0) && (ppfdCandidate->cColorBits != dm.dmBitsPerPel)) {
+ 		continue;
+ 	}
+ 
  	/*
  	** See if candidate is better than the previous best choice
  	*/
