from Mesa-5.0.x.

Index: xc/extras/Mesa/src/X86/common_x86_asm.S
diff -u xc/extras/Mesa/src/X86/common_x86_asm.S:1.16 xc/extras/Mesa/src/X86/common_x86_asm.S:1.17
--- Mesa/src/X86/common_x86_asm.S:1.16	Mon Dec 16 11:18:33 2002
+++ Mesa/src/X86/common_x86_asm.S	Sun Sep 28 15:17:08 2003
@@ -59,11 +59,6 @@
 GLNAME( found_intel ):	STRING( "Genuine Intel processor found\n\0" )
 GLNAME( found_amd ):	STRING( "Authentic AMD processor found\n\0" )
 
-#ifdef USE_KATMAI_ASM
-GLNAME( katmai_test_dummy ):
-	D_LONG 0x3f800000, 0x3f800000, 0x3f800000, 0x3f800000
-#endif
-
 
 	SEG_TEXT
 
@@ -217,7 +212,13 @@
 	LDMXCSR	( REGOFF( -8, EBP ) )
 
 	XORPS	( XMM0, XMM0 )
-	MOVUPS	( GLNAME( katmai_test_dummy ), XMM1 )
+
+	PUSH_L	( CONST( 0x3f800000 ) )
+	PUSH_L	( CONST( 0x3f800000 ) )
+	PUSH_L	( CONST( 0x3f800000 ) )
+	PUSH_L	( CONST( 0x3f800000 ) )
+
+	MOVUPS	( REGIND( ESP ), XMM1 )
 
 	DIVPS	( XMM0, XMM1 )
 

