diff -urNp Mesa-3.4.2/src/FX/fxdos.c Mesa-3.4.2x/src/FX/fxdos.c
--- Mesa-3.4.2/src/FX/fxdos.c	2016-03-18 18:01:10.000000000 +0200
+++ Mesa-3.4.2x/src/FX/fxdos.c	2016-03-21 22:41:50.000000000 +0200
@@ -55,6 +55,8 @@
 #include "GL/gl.h"
 #include "GL/fxmesa.h"
 
+extern void GLAPIENTRY gl3DfxSetPaletteEXT(GLuint*);
+
 #if 0
 /* copied from fxwgl.c */
 
@@ -88,6 +90,7 @@ static struct __extensions__   ext[] = {
    { (fxMesaProc)glCopyTexSubImage3DEXT,		"glCopyTexSubImage3DEXT"	},
    { (fxMesaProc)glTexImage3DEXT,			"glTexImage3DEXT"		},
    { (fxMesaProc)glTexSubImage3DEXT,			"glTexSubImage3DEXT"		},
+/* { (fxMesaProc)gl3DfxSetPaletteEXT,			"gl3DfxSetPaletteEXT"		},*/
    { (fxMesaProc)glColorTableEXT,			"glColorTableEXT"		},
    { (fxMesaProc)glColorSubTableEXT,			"glColorSubTableEXT"		},
    { (fxMesaProc)glGetColorTableEXT,			"glGetColorTableEXT"		},
diff -urNp Mesa-3.4.2/src/FX/fxwgl.c Mesa-3.4.2x/src/FX/fxwgl.c
--- Mesa-3.4.2/src/FX/fxwgl.c	2016-03-18 18:01:10.000000000 +0200
+++ Mesa-3.4.2x/src/FX/fxwgl.c	2016-03-21 22:41:50.000000000 +0200
@@ -108,7 +108,7 @@ static struct __extensions__   ext[] = {
    { (PROC)glCopyTexSubImage3DEXT,		"glCopyTexSubImage3DEXT"	},
    { (PROC)glTexImage3DEXT,			"glTexImage3DEXT"		},
    { (PROC)glTexSubImage3DEXT,			"glTexSubImage3DEXT"		},
-   { (PROC)gl3DfxSetPaletteEXT,			"3DFX_set_global_palette"	},
+/* { (PROC)gl3DfxSetPaletteEXT,			"gl3DfxSetPaletteEXT"		},*/
    { (PROC)glColorTableEXT,			"glColorTableEXT"		},
    { (PROC)glColorSubTableEXT,			"glColorSubTableEXT"		},
    { (PROC)glGetColorTableEXT,			"glGetColorTableEXT"		},
diff -urNp Mesa-3.4.2/src/Makefile.dxe Mesa-3.4.2x/src/Makefile.dxe
--- Mesa-3.4.2/src/Makefile.dxe	2016-03-18 18:01:10.000000000 +0200
+++ Mesa-3.4.2x/src/Makefile.dxe	2016-03-21 22:41:50.000000000 +0200
@@ -14,6 +14,9 @@ GL_LIB = libgl.a
 GL_DXE = gl.dxe
 GL_IMP = libigl.a
 LIBNAME="fxMesa-v3.4"
+DXE_EXPORTS= -E _gl -X _gl_ -X _glb
+DXE_EXPORTS+= -E _fxMesa -E _fxQueryHardware -E _fxCloseHardware -E _fxGetScreenGeometry
+DXE_DEPS= -P glide3x.dxe
 
 CC = gcc
 AR = ar
@@ -175,7 +178,7 @@ $(LIBDIR)/$(GL_LIB): $(LIBDIR) $(OBJECTS
 
 $(LIBDIR)/$(GL_DXE) $(LIBDIR)/$(GL_IMP): $(LIBDIR) $(OBJECTS)
 $(LIBDIR)/$(GL_DXE) $(LIBDIR)/$(GL_IMP): glide3x.dxe
-	-dxe3gen -o $(LIBDIR)/$(GL_DXE) -Y $(LIBDIR)/$(GL_IMP) -D $(LIBNAME) -E _gl -X _gl_ -E _fxMesa -E _fxQuery -E _fxClose -P glide3x.dxe -U $(OBJECTS)
+	-dxe3gen -o $(LIBDIR)/$(GL_DXE) -Y $(LIBDIR)/$(GL_IMP) -D $(LIBNAME) $(DXE_EXPORTS) $(DXE_DEPS) -U $(OBJECTS)
 
 # dxe3gen needs glide3x.dxe when generating the import lib libigl.a.
 # so we generate a dummy glide3x.dxe to cover common glide3x exports
